src/manage_nsfs/nc_lifecycle.js

```diff
@@ -735,6 +741,32 @@ async function _call_op_and_update_status({ bucket_name = undefined, rule_id = u
     }
 }
 
+/**
+ * init the bucket status object statuses if they don't exist
+ * @param {string} bucket_name
+ * @returns {Object} created or existing bucket status
+ */
+function init_bucket_status(bucket_name) {
+    lifecycle_run_status.buckets_statuses[bucket_name] ??= {};
+    lifecycle_run_status.buckets_statuses[bucket_name].rules_statuses ??= {};
+    lifecycle_run_status.buckets_statuses[bucket_name].bucket_process_times = {};
+    lifecycle_run_status.buckets_statuses[bucket_name].bucket_stats = {};
+    return lifecycle_run_status.buckets_statuses[bucket_name];
+}
+
+/**
+ * init the rule status object statuses if they don't exist
+ * @param {string} bucket_name
+ * @param {string} rule_id
+ * @returns {Object} created or existing rule status
+ */
+function init_rule_status(bucket_name, rule_id) {
+    lifecycle_run_status.buckets_statuses[bucket_name].rules_statuses[rule_id] ??= {};
```

Comment:
state is not mentioned here but it's a part of both rule/bucket status
we probably should init it as well, although on --continue you don't want to nullify it

Reply:
we can. but it means we will need to run the init before that. I will change it